pycalphad/core/minimizer.pyx

```diff
@@ -377,6 +377,9 @@ cdef class SystemSpecification:
             state.increment_phase_metastability_counters()
             if not phases_changed:
                 advance_state(self, state, eq_soln, step_size)
+        if state.free_stable_compset_indices.shape[0] > self.max_num_free_stable_phases:
+            # Gibbs phase rule violation in solution
+            converged = False
```

Comment:
The idea is that we would hit this if the other change to MIN_PHASE_AMOUNT was not sufficient to remove enough phases violating Gibbs phase rule? So the other change justifies not coverage?

Reply:
Correct. The challenge with writing a test to cover that branch is that, ideally, that branch should be unreachable. I don't know how to write a test case that reaches that branch; if I did, that would itself be a bug. The new MIN_PHASE_AMOUNT logic has addressed the only known case, and this branch is defensive programming. Unfortunately we don't currently have a way to bubble specific error messages up with the convergence state.